docs/operational_guide/replication_and_deployment_in_zones.md

```diff
@@ -0,0 +1,57 @@
+
+### Replication
+
+It is highly recommended that you use a replication factor of at least 3 for any M3DB deployment. This is due to the default write and read consistency levels requiring a majority of hosts to be available to complete an operation successfully. For more information see the documentation concerning tuning [availability, consistency and durability](availability_consistency_durability.md).
+
+### Upgrading hosts in a deployment
+
+It should be noted that when a host is deployed and it is preparing to serve reads by bootstrapping, the shards are available for writes but not available for reads.
+
+There is a small window during an upgrade between when the process is stopped and then started again where the node is unavailable for writes too.
+
+## Deployment in a single zone
+
+For deployment in a single zone, we recommend you set the `isolationGroup` to the name of the rack a host resides or another logical unit that separates groups of hosts in your zone.
+
+In this configuration hosts are distributed shards of data that will not be placed more than once in the same rack or logical unit you define, which allows you to lose an entire unit at a time that may contain multiple hosts and still stay available since it is guaranteed to only affect one replica of data.
+
+Here's what a deployment in a single zone looks like, in this example the cluster has 3 shards that are spread over 4 racks:
+
+![Replication Single Zone](replication_single_zone.png)
+
+Typically deployments have many more than 3 shards, but this is a simple example that depicts how losing a single rack still maintains availability of two of the three replicas of each shard.
+
+## Deployment across multiple availability zones in a region
+
+For deployment in a region, we recommend you set the `isolationGroup` to the name of the availability zone a host resides.
+
+In this configuration hosts are distributed shards of data that will not be placed more than once in the same availability zone.
+
+Here's what a deployment in a region looks like, in this example the cluster has 4 shards that are spread over 3 availability zones:
+
+![Replication Region](replication_region.png)
+
+Typically deployments have many more than 4 shards, but this is a simple example that depicts how losing a single availability zone still maintains availability of two of the three replicas of each shard.
+
+<!-- **Note:** Placement of shards in a replica do not mirror other replicas and shards are placed in a way to best load balance shards across available hosts in a single isolation group. This could mean that hosts in an availability zones that have less hosts will do more work than those in an availability zone with more hosts. -->
```

Comment:
The mirroring thing might make things more confusing. Mirror deployments are uncommon in most setups that I've seen so I think leaving out any reference to it might be less confusing.

Also I think we can rephrase to: "M3DB will do its best to distribute shards evenly among the availability zones while still taking each individual node's weight into account, but if some of the availability zones have less available hosts than others then each host in that zone will be responsible for more shards than hosts in the other zones and will thus be subjected to heavier load." or something like that

Also should we move this note out of this section and make it more general? It applies if you have an uneven number of hosts in your rack distribution or an uneven number of hosts in your multi region setup too

Reply:
Yeah I removed the mirroring part and have moved it under the "replication" header.  I'll use the suggested text you have, slightly better than my revised one, TY!